src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/unusedlocalvariable/InputUnusedLocalVariable.java

```diff
@@ -0,0 +1,283 @@
+    }
+
+    public void testMultipleIdentifiers() {
+        int[] arr = {1};
+        int a = 21;
+        arr[0] = a;
+        Class<Integer> b = int.class; // violation
+        @SuppressWarnings("unused")
+        int c = 41; // violation
+    }
+
+    public void testChainedCalls() throws InterruptedException {
+        Obj p; // violation
+        Obj q = null; // ok
+        p = q.foo().p;
+        if(s instanceof String) {
+        }
+        Boolean r = false;
+        while(r.booleanValue() != true) {
+        }
+        int a = 13, b= 21;
+        q.getClass().wait(a,b);
+        int f = 12; // violation
+        Predicate<String> obj = InputUnusedLocalVariable::f;
+        obj.test("test");
+        int foo = 12; // violation
+        foo(q);
+    }
+
+    class SubClass extends InputUnusedLocalVariable {
+
+        void testSuperKeyword() {
+            int a = 12; // violation
+            a = super.a;
+        }
+    }
+
+    class Obj {
+        Obj p;
+        Obj foo() {
+            return new Obj();
+        }
+    }
+
+    public void testStatementsWithoutSlistToken() {
+        int k = 1;
+        for (int i = 0; k < 1; i++) // violation
+            i++;
+
+        for (int j = 0; ; ) { // violation
+            j++;
+            break;
+        }
+
+        ArrayList<String> obj = new ArrayList<>();
+
+        for (String s : obj)
+            System.out.println(s);
+    }
+
+    public void testChainedCallsWithNewKeyword() {
+        Object a; // violation
+        Object b; // violation
+        a = foo(new c.b(new d.a()));
+        a = foo(new b().new a());
+    }
+
+    public void testBooleanExpressions() {
+        boolean b = false; // ok
+        if (!b) {
+        }
+        boolean b1 = true; // ok
+        if (b1) {
+        }
+        boolean a; // violation
+        if ((a = true) != false) {
+        }
+        boolean j; // violation
+        if (j = true) {
+        }
+        boolean k = true, l = false;
+        if (k && !l) {
+        }
+    }
+
+    public void testTryWithResources() {
+        try (BufferedReader br =
+                new BufferedReader(new FileReader("someFile.txt"))) { // ok
+            br.readLine();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    static class Test implements a {
+
+        public void testMethodReference() {
+            int b = 12; // violation
+            int Test = 12; // violation
+            a obj = Test::new; // violation
+            Test ab = new Test();
+            Predicate<String> a = ab::b;
+            a.test("abc");
+        }
+
+        private boolean b(String s) {
+            return true;
+        }
+
+        public static boolean ba(String s) {
+            return true;
+        }
+
+        @Override
+        public void method() {
+        }
+    }
+
+    public void testDifferentAssignments() {
+        int a = 12; // violation
+        a = 13;
+        int b = 13; // ok
+        b /= 12;
```

Comment:
Why is `b` ok but `a` is basically the same thing and has a violation? Same with c and h?

Reply:
`a` is simply assigned the value `13` but `b` is not simply assigned, `b = b/12` so `b` will be considered as used, same for others.